trifid/redirect.js

```diff
@@ -1,22 +1,64 @@
+// @ts-check
+const { fetchBuilder, MemoryCache } = require('node-fetch-cache')
+
+/**
+ * Perform some redirections.
+ *
+ * @returns {import('express').RequestHandler}
+ */
 function factory () {
-  return (req, res, next) => {
+  const cachedFetch = fetchBuilder.withCache(new MemoryCache({
+    ttl: 60 * 60 * 1000, // 1 hour
+  }))
+
+  return async (req, res, next) => {
+    const requestPath = req.path
+
     // ignore health request from the cluster
-    if (req.path === '/health') {
+    if (requestPath === '/health') {
       return next()
     }
 
-    if (req.path.match(/^\/[^/]+$/)) {
+    if (requestPath.match(/^\/[^/]+$/)) {
       return res.redirect('/')
     }
 
-    if (req.path.match(/^\/meta\/[^/]+$/)) {
+    if (requestPath.match(/^\/meta\/[^/]+$/)) {
       return res.redirect('/meta/')
     }
 
-    if (req.path.match(/^\/relation\/[^/]+$/)) {
+    if (requestPath.match(/^\/relation\/[^/]+$/)) {
       return res.redirect('/relation/')
     }
 
+    // Redirect to a version of a shape (for routes: `/vX.Y.Z/shape/NAME` and `/latest/shape/NAME`)
+    const extensionPath = requestPath.endsWith('.ttl') ? '' : '.ttl'
```

Comment:
I'm only not convinced that we need to support `.ttl` at all. The canonical shape identifiers should be without extensions